### ISSUE TITLE
Timezone support ?

### ISSUE THREAD

**Issue by yellowiscool** [opening-hours/opening_hours.js#269]
How could I specify the timezone ?

**Comment by HolgerJeromin** [comment]
What is your use case?

**Comment by yellowiscool** [comment]
I will use a story: 

I live in Norway and my web browser uses the `Europe/Oslo` timezone. My grandmother is in holidays in Martinique (France), which has the timezone `America/Martinique`.  She calls me during the night, her car has an issue and she asks me to find an open garage.

I go to an openstreetmap application which use opening_hours.js to say whether a shop is open or not. I find a garage, with the opening_hours tag `Mo-Fr 09:00-17:00`. It is supposed to be open, however because opening_hours.js doesn't support timezone, it shows as closed on my openstreetmap application.

**Comment by StephanGeorg** [comment]
Yes, I came across the same problem and solved it with the following solution by using [date-timezones](https://www.npmjs.com/package/date-timezones) to generate a date object in a given timezone:

```
import dateTimezones from 'date-timezones';

const value = 'Mo-Fr 09:00-17:00';  // Opening hours of location
const timezone = 'America/Martinique';  // Timezone of location

const oh = new OpeningHours(value, {}, {});

const locationTime = dateTimezones(timezone);
const isOpen = oh.getState(locationTime);
```
I don't think you should implement it into opening_hours.js because getState, getNextChange, ... expects a date object which can be generated for your needs.

**Comment by ypid** [comment]
Thanks, I also agree that the library itself does not need to be timezone aware. But the map should.

**Comment by andidev** [comment]
@StephanGeorg 
Is your solution DST safe? What about `getOpenIntervals`?

**Comment by StephanGeorg** [comment]
> Is your solution DST safe?

Hm? Good question, @ypid?

> What about getOpenIntervals?

Yes, it accepts a date object like all other methods

```
var oh = new opening_hours('We 12:00-14:00');

var from = new Date("01 Jan 2012");
var to   = new Date("01 Feb 2012");

// high-level API
{
    var intervals = oh.getOpenIntervals(from, to);
    ...
```

**Comment by andidev** [comment]
@StephanGeorg
Yes that is partially true, the problem is that the dates return from `getOpenIntervals` will be in the incorrect timezone.

**Comment by StephanGeorg** [comment]
But does the timezone really matter in that case? `getOpenIntervals` returns opening intervals from starting date till end date. Why would you add a timezone? Or do I miss anything?

**Comment by andidev** [comment]
Well it matters if you want to show the opening hours in locale time (e.g. Norwegian time as in the mentioned example). Though maybe that is not usually what you wan't to show the user so maybe you are right that it does not matter.

**Comment by atrick-speedline** [comment]
Here's an example of where this is an issue. If I'm in southern British Columbia (uses DST) on March 9 at noon, and I want to see how long it will be until a shop opens at 8 in northern British Columbia (no DST), it will tell me 19 hours, even though it should tell me 20 hours.

**Comment by Its-Just-Nans** [comment]
On my side I did

```ts
import tzlookup from '@photostructure/tz-lookup'

// Fake the local date
const getLocateDate = (lat?: number, lon?: number) => {
    if (!lat || Number.isNaN(lat) || !lon || Number.isNaN(lon)) {
        return new Date()
    }
    const localDateString = (new Date()).toLocaleString('en-US', {
        timeZone: tzlookup(lat, lon),
    })
    return new Date(localDateString)
}


const oh = new opening_hours(...)

const localDate = getLocateDate(lat, lon)
const isOpen = oh.getState(localDate)
const nextChange = oh.getNextChange(localDate)
```
